dff/core/engine/core/context.py

```diff
@@ -174,8 +173,12 @@ def add_label(self, label: NodeLabel2Type):
         last_index = get_last_index(self.labels)
         self.labels[last_index + 1] = label
 
-    @validate_arguments  # todo: use set instead of a list
```

Comment:
Sorry, I meant tuple instead of set / list because that would make the default `field_names` argument immutable.

Reply:
I think, we can skip this TODO